src/modules/wallets/RPCWallet.js

```diff
@@ -688,6 +704,85 @@ class RPCWallet {
     // Return the TXID of the transaction
     return broadcastTX.result
   }
+
+  /**
+   * Subscribe a callback function to be run when all TXIDs are confirmed.
+   * @param  {Array<String>} response.txids - An Array of TXIDs
+   * @param  {OIPRecord} response.record - The OIPRecord that was published (to be returned to the Callback)
+   * @param {function} options.onConfirmation - A function to run once the transaction recieves a confirmation that it was included in a block.
+   * @param {String} options.onConfirmationRef - A reference string that should be returned in the onConfirmation function (useful for ID's).
+   */
+  onConfirmation ({ txids, record }, options) {
+    // If we do not have an onConfirmation subscription, exit the function
+    if (!options || !options.onConfirmation || typeof options.onConfirmation !== 'function') { return }
+
+    // Add this subscription to the tracked subscriptions
+    this.onConfirmationSubscriptions[record.getTXID()] = {
+      txids,
+      record,
+      options
+    }
+  }
+
+  /**
+   * Check if any of the confirmation subscriptions should be run because TXIDs got confirmed
+   * @return {Promise} Returns a promise that resolves once all of the available confirmation callbacks have been run
+   */
+  async checkForConfirmations () {
+    for (let subscription in this.onConfirmationSubscriptions) {
+      let { txids, record, options } = this.onConfirmationSubscriptions[subscription]
+
+      // Check if we are still waiting to be confirmed
+      let waitingForConfirmation = false
+      for (let txid of txids) {
+        if (this.unconfirmedTxids.includes(txid)) {
+          waitingForConfirmation = true
+        }
+      }
+
+      if (!waitingForConfirmation) {
+        try {
+          await options.onConfirmation(record, txids, options.onConfirmationRef)
+        } catch (e) {
+          console.warning(`[RPC Wallet] Error when running onConfirmation function in subscription: `, e)
```

Comment:
should anything happen if an error is thrown?

Reply:
We don't want to fail and crash currently, just log that there was an error for now